R/dbplyr.R

```diff
@@ -76,257 +37,140 @@ tbl.Pool <- function(src, from, ..., vars = NULL, con = NULL) {
-  dplyr::db_query_rows(db_con, sql = sql, ...)
-}
+      !!recall
+    })
 
-#' @rdname dplyr-db-methods
-db_rollback.Pool <- function(con, ...) {
-  db_con <- poolCheckout(con)
-  on.exit(poolReturn(db_con))
-  dplyr::check_dbplyr()
-  dplyr::db_rollback(db_con, ...)
-}
-
-#' @rdname dplyr-db-methods
-db_save_query.Pool <- function(con, sql, name, temporary = TRUE, ...) {
-  stopIfTemporary(temporary)
-  db_con <- poolCheckout(con)
-  on.exit(poolReturn(db_con))
-  dplyr::check_dbplyr()
-  dplyr::db_save_query(db_con, sql = sql, name = name,
-                       temporary = temporary, ...)
-}
-
-#' @rdname dplyr-db-methods
-db_write_table.Pool <- function(con, table, types, values,
-                                temporary = FALSE, ...) {
-  stopIfTemporary(temporary)
-  db_con <- poolCheckout(con)
-  on.exit(poolReturn(db_con))
-  dplyr::check_dbplyr()
-  dplyr::db_write_table(db_con, table = table, types = types,
-                        values = values, temporary = temporary, ...)
-}
-
-#' @rdname dplyr-db-methods
-sql_escape_string.Pool <- function(con, x) {
-  db_con <- poolCheckout(con)
-  on.exit(poolReturn(db_con))
-  dplyr::check_dbplyr()
-  dplyr::sql_escape_string(db_con, x = x)
-}
-
-#' @rdname dplyr-db-methods
-sql_join.Pool <- function(con, x, y, vars, type = "inner", by = NULL, ...) {
-  db_con <- poolCheckout(con)
-  on.exit(poolReturn(db_con))
-  dplyr::check_dbplyr()
-  dplyr::sql_join(db_con, x = x, y = y, vars = vars, type = type,
-                  by = by, ...)
-}
-
-#' @rdname dplyr-db-methods
-sql_select.Pool <- function(con, select, from, where = NULL,
-                            group_by = NULL, having = NULL, order_by = NULL, limit = NULL,
-                            distinct = FALSE, ...) {
-  db_con <- poolCheckout(con)
-  on.exit(poolReturn(db_con))
-  dplyr::check_dbplyr()
-  dplyr::sql_select(db_con, select = select, from = from,
-                    where = where, group_by = group_by, having = having,
-                    order_by = order_by, limit = limit, distinct = distinct, ...)
-}
-
-#' @rdname dplyr-db-methods
-sql_semi_join.Pool <- function(con, x, y, anti = FALSE, by = NULL, ...) {
-  db_con <- poolCheckout(con)
-  on.exit(poolReturn(db_con))
-  dplyr::check_dbplyr()
-  dplyr::sql_semi_join(db_con, x = x, y = y, anti = anti, by = by, ...)
-}
-
-random_table_name <- function(n = 10) {
-  paste0(sample(letters, n, replace = TRUE), collapse = "")
-}
-
-#' @rdname dplyr-db-methods
-sql_subquery.Pool <- function(con, from,
-                              name = random_table_name(), ...) {
-  db_con <- poolCheckout(con)
-  on.exit(poolReturn(db_con))
-  dplyr::check_dbplyr()
-  dplyr::sql_subquery(db_con, from = from, name = name, ...)
-}
-
-#' @rdname dplyr-db-methods
-sql_translate_env.Pool <- function(con) {
-  db_con <- poolCheckout(con)
-  on.exit(poolReturn(db_con))
-  dplyr::check_dbplyr()
-  dplyr::sql_translate_env(db_con)
+    new_function(args, body, env = ns_env("pool"))
```

Comment:
If possible, I think it's better not to override the environment. This way the code above can be read in (lexical) context.